javaparser-core/src/main/java/com/github/javaparser/ast/Node.java

```diff
@@ -65,11 +62,13 @@
 	public Node() {
 	}
 
+	public Node(Position begin, Position end) {
+		this.begin = begin;
```

Comment:
As the positions aren't mandatory, throwing an exception would indeed be mean...